MantidQt/CustomInterfaces/src/Tomography/TomographyIfacePresenter.cpp

```diff
@@ -146,6 +150,75 @@ void TomographyIfacePresenter::notify(
   }
 }
 
+void TomographyIfacePresenter::processRunExternalProcess() {
+  // TODO unify with processRunRecon()
+  // read the cached variables in view
+  const std::string &cachedExec = m_view->getCachedExecutable();
+  const std::vector<std::string> &cachedArgs = m_view->getCachedArguments();
+
+  std::string runnable;
+  std::vector<std::string> args;
+  std::string allOpts;
+  // still prepare submission arguments to get the defaults
+  const bool local = true;
+  prepareSubmissionArguments(local, runnable, args, allOpts);
+
+  // append the additional args for now
+  for (const auto &arg : cachedArgs) {
+    args.emplace_back(arg);
+    allOpts += arg;
+  }
+
+  // if no external provided, use the default one from settings
+  setupAndRunLocalExternalProcess(!cachedExec.empty() ? cachedExec : runnable,
```

Comment:
Could we split this operation out: `!cachedExec.empty() ? cachedExec : runnable` as there are several operations running within a function call meaning you have to read it very carefully